.github/workflows/nightly-rosetta-pax-build.yaml

```diff
@@ -0,0 +1,74 @@
+name: Nightly Rosetta Paxml build and test
+
+on:
+  workflow_run:
+    workflows: [Nightly Pax build]
+    types: [completed]
+    branches: [main]
+  workflow_dispatch:
+    inputs:
+      PUBLISH:
+        type: boolean
+        description: Publish dated images and update the 'latest' tag?
+        default: false
+        required: false
+
+env:
+  TARGET: rosetta
+  DOCKER_REGISTRY: ghcr.io/nvidia
+
+permissions:
+  contents: read  # to fetch code
+  actions:  write # to cancel previous workflows
+  packages: write # to upload container
+
+jobs:
+
+  metadata:
+    if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-22.04
+    outputs:
+      BUILD_DATE: ${{ steps.date.outputs.BUILD_DATE }}
+    steps:
+      - name: Set build date
+        id: date
+        shell: bash -x -e {0}
+        run: |
+          BUILD_DATE=$(TZ='US/Los_Angeles' date '+%Y-%m-%d')
+          echo "BUILD_DATE=${BUILD_DATE}" >> $GITHUB_OUTPUT
+
+  build:
+    if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || github.event_name == 'workflow_dispatch'
+    needs: metadata
+    uses: ./.github/workflows/_build_rosetta.yaml
+    with:
+      BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE }}
+      BASE_LIBRARY: pax
+      BASE_IMAGE: ${{ env.DOCKER_REGISTRY }}/pax:latest
+    secrets: inherit
+
+  test:
```

Comment:
So this suggestion and https://github.com/NVIDIA/JAX-Toolbox/pull/65#discussion_r1196798672 are related:

The issue I ran into with matrix builds here is that I want a "matrix build" to next trigger a "matrix test". It seemed like there wasn't great GH actions support for this. I've seen a workaround where someone passes a json artifact as the output of a "matrix build" and the "matrix test" consumes it with the grammar `strategy.matrix: ${{ fromJSON(needs.build.outputs.matrix) }}`. But IIUC, the test jobs need to wait until ALL of the matrix jobs finish to start testing.

So the workaround was just to combine build and test to simplify things, since the test can start right after the build, and I wouldn't have to rewrite a lot of the same yaml file to have a concrete `nightly_rosetta_${BASE_LIBRARY}_test.yaml`. The other benefit is the one you mentioned where build/test/publish are all in the same workflow; so (in the future) we can publish the images and badges contingent on the tests succeeding. Also, I don't think this is a concern now, but it seems like this will postpone us needing to address `workflow_run`'s [3 workflow limit](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run:~:text=You%20can%27t%20use%20workflow_run%20to%20chain%20together%20more%20than%20three%20levels%20of%20workflows.)

I can re-separate build and test, but having dependent matrix builds seems like more work than they're worth. I'm curious if you know of a better solution here.

Reply:
That's a really good point. Let's keep building and testing together here, while I prioritize the badge creation process?